src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -2337,4 +2430,121 @@ RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
 #endif // backend
 }
 
+/******************** fog ********************/
+
+RppStatus rppt_fog_gpu(RppPtr_t srcPtr,
+                       RpptDescPtr srcDescPtr,
+                       RppPtr_t dstPtr,
+                       RpptDescPtr dstDescPtr,
+                       Rpp32f *gammaTensor,
+                       RpptROIPtr roiTensorPtrSrc,
+                       RpptRoiType roiType,
+                       rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+
+    RpptInterpolationType interpolationType = RpptInterpolationType::NEAREST_NEIGHBOR;
+    RpptDesc fogMaskSrcDesc;
+    RpptDescPtr fogMaskSrcDescPtr = &fogMaskSrcDesc; // Fog original mask source description pointer
+    set_fog_mask_descriptor(fogMaskSrcDescPtr, 2, FOG_MAX_HEIGHT, FOG_MAX_WIDTH, 1);  
+
+    RpptDesc fogMaskDstDesc;
+    RpptDescPtr fogMaskDstDescPtr = &fogMaskDstDesc; //Fog resized mask destination description pointer
+    set_fog_mask_descriptor(fogMaskDstDescPtr, 2, srcDescPtr->h, srcDescPtr->w, 1);
+
+    RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *>(rpp::deref(rppHandle).GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem);
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + 2);
+
+    for(Rpp32s i = 0; i < 2; i++)
+    {
+        internalDstImgSizes[i] = {srcDescPtr->w, srcDescPtr->h};
+        internalRoiTensorPtrSrc[i].xywhROI = {0, 0, 1920, 1080};
+    }
+    rppSetBatchSize(rppHandle,2);                   // Setting batch size to 2 for computing resizing for alpha and intensity
+
+    Rpp32u maskSize = FOG_MAX_HEIGHT * FOG_MAX_WIDTH;
+    Rpp32u maskSizeInBytes = maskSize * sizeof(Rpp32f);
+
+    Rpp32f *d_fogAlphaMaskPtr, *d_fogIntensityMaskPtr;
+    d_fogAlphaMaskPtr = reinterpret_cast<Rpp32f*>(rpp::deref(rppHandle).GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+
+    Rpp32f *d_resizedFogAlphaMaskPtr, *d_resizedFogIntensityMaskPtr;
+    d_resizedFogAlphaMaskPtr = reinterpret_cast<Rpp32f*>(d_fogAlphaMaskPtr + (2 * maskSize));
+
+    CHECK_RETURN_STATUS(hipMemcpyAsync(d_fogAlphaMaskPtr, &fogMask_1920_1080[0], maskSizeInBytes * 2, hipMemcpyHostToDevice,rpp::deref(rppHandle).GetStream())); // Copying fog alpha mask from host to device asynchronously
+    rppt_resize_gpu(d_fogAlphaMaskPtr, fogMaskSrcDescPtr, d_resizedFogAlphaMaskPtr, fogMaskDstDescPtr, internalDstImgSizes, interpolationType, internalRoiTensorPtrSrc, roiType, rppHandle); // Performing GPU resize operation on fog alpha and intensity mask
+    hipStreamSynchronize(rpp::deref(rppHandle).GetStream());
+    d_resizedFogIntensityMaskPtr = d_resizedFogAlphaMaskPtr + (srcDescPtr->h * srcDescPtr->w);
+
+    rppSetBatchSize(rppHandle,srcDescPtr->n);       // Resetting the batch size to match the source's batch size
```

Comment:
Same comment
Add space after ,
```suggestion
    rppSetBatchSize(rppHandle, srcDescPtr->n);       // Resetting the batch size to match the source's batch size
```